test/framework/src/main/java/org/elasticsearch/cluster/ESAllocationTestCase.java

```diff
@@ -87,13 +87,19 @@ public static MockAllocationService createAllocationService(Settings settings, C
         );
     }
 
-    // TODO these tests should work with both allocators
     private static final Settings BALANCED_ALLOCATOR_BY_DEFAULT = Settings.builder()
         .put(SHARDS_ALLOCATOR_TYPE_SETTING.getKey(), BALANCED_ALLOCATOR)
         .build();
 
+    private static final Settings DESIRED_BALANCE_ALLOCATOR_BY_DEFAULT = Settings.builder()
+        .put(SHARDS_ALLOCATOR_TYPE_SETTING.getKey(), DESIRED_BALANCE_ALLOCATOR)
+        .build();
+
     private static ShardsAllocator createShardsAllocator(Settings settings) {
-        return switch (SHARDS_ALLOCATOR_TYPE_SETTING.get(settings, BALANCED_ALLOCATOR_BY_DEFAULT)) {
+        return switch (SHARDS_ALLOCATOR_TYPE_SETTING.get(
+            settings,
+            randomFrom(BALANCED_ALLOCATOR_BY_DEFAULT, DESIRED_BALANCE_ALLOCATOR_BY_DEFAULT)
+        )) {
```

Comment:
Do we want to keep allocator configurable by the test settings? May be simplify to:

```suggestion
        return switch (randomFrom(BALANCED_ALLOCATOR_BY_DEFAULT, ESIRED_BALANCE_ALLOCATOR_BY_DEFAULT)) {
```